lib/mpl_toolkits/mplot3d/art3d.py

```diff
@@ -182,14 +179,13 @@ def set_3d_properties(self, z=0, zdir='z', axlim_clip=False):
     @artist.allow_rasterization
     def draw(self, renderer):
         if self._axlim_clip:
-            xs, ys, zs = _viewlim_mask(self._x, self._y, self._z, self.axes)
-            position3d = np.ma.row_stack((xs, ys, zs)).ravel().filled(np.nan)
+            mask = _viewlim_mask(self._x, self._y, self._z, self.axes)
+            pos3d = np.array([self._x, self._y, self._z], dtype=float)
```

Comment:
minor nit while looking at this, but you could bring `pos3d` creation above the if block and remove the else here.